### ISSUE TITLE
data localization not working just appeard but there is no impact done

### ISSUE THREAD

**Issue by ahmedzedo** [OrchardCoreContrib/OrchardCoreContrib.Modules#115]
I have installed the Nuget Package and after using the data resource section for content types and content fields the values didn't appear after setting the translation so I can't see any impact of data localization are there any prerequisites that I should do to the Orchard site and content feel with translation impact

**Comment by hishamco** [comment]
You can check the [README](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/main/src/OrchardCoreContrib.DataLocalization/README.md) file

**Comment by hishamco** [comment]
Please a GIF screenshot

**Comment by ahmedzedo** [comment]
@hishamco 
![image](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/assets/24712589/c730a508-52f1-4f32-a076-0721b749c3a1)
![image](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/assets/24712589/da21b55c-38c3-49dc-8e0f-70074b72b45a)
![image](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/assets/24712589/13d724b0-d712-4417-a02e-8ba00d6910d2)

**Comment by hishamco** [comment]
Yes this can't be localized because the original OC doesn't use a Data Localization module, but if you inject the data localizer into your view you will be easy to localize them

**Comment by ahmedzedo** [comment]
@hishamco I'm using the Orchard website with the headless recipe I just need the names of content types and custom fields to appear based on the current language in the admin panel

**Comment by hishamco** [comment]
Regardless of what methodology you are using OC is unable to show the Content Types in Arabic because they don't implement dynamic localization yet, but I will struggle to accelerate the PR I did a long time ago

**Comment by hishamco** [comment]
One possible option is to override Orchard Core pages

**Comment by ahmedzedo** [comment]
@hishamco what are the benefits of the [OrchardCoreContrib?DataLocalization] Nuget package that I installed and enabled on my website while doesn't work yet, you mean that the package till now appeared as a module but still doesn't work

**Comment by hishamco** [comment]
It's working as expected, you can take the benefits if you are using the data localization service inside OC, this was originally a PoC for dynamic data localization

**Comment by ahmedzedo** [comment]
@hishamco Actually, I watched you at the conference demonstrate it, but I followed the steps I installed the Nuget package and I enabled the data localization and I used it as the image that I sent before but it doesn't change anything, so do you have an idea why it isn't working with me despite it appears and save the values properly 
![image](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/assets/24712589/4cdafb19-32f4-43fc-8bcd-68d0fc7e9016)

**Comment by hishamco** [comment]
If the values save properly, it means everything working as expected, could you please add a razor view and inject `IDataLocalizer`

**Comment by ahmedzedo** [comment]
@hishamco I have injected IDataLocalizer in razor page on my website I reran the project but there is no change occurred still the data localizer package doesn't work

**Comment by hishamco** [comment]
Did you enable the feature? Did you specify the context?

**Comment by hishamco** [comment]
You need to specify the context for each localized string because it could be `Content Type`, or `Content Field` .. etc

I might need to add functional test, also I might demo the data localization again in the Orchard Core Meeting

**Comment by ahmedzedo** [comment]
@hishamco Yes,  you already divided them into two sections. I followed the demo as you demonstrated so, I think there is an issue. If there is another version that may be working fine. please, let me know how can I download this version

**Comment by hishamco** [comment]
Let me check

**Comment by hishamco** [comment]
Have a look to the unit tests [here](https://github.com/OrchardCoreContrib/OrchardCoreContrib/blob/main/test/OrchardCoreContrib.Localization.Tests/Data/DataLocalizerTests.cs)

You can share a minimal repo with the reproduced issue

**Comment by hishamco** [comment]
@ahmedzedo I didn't from awhile, please let me know if the issue exists or I will close this
